lib/chef/knife/ec2_server_create.rb

```diff
@@ -824,7 +824,7 @@ def create_server_def
           :groups => config[:security_groups],
           :security_group_ids => locate_config_value(:security_group_ids),
           :flavor_id => locate_config_value(:flavor),
-          :key_name => Chef::Config[:knife][:aws_ssh_key_id],
+          :key_name => config[:ssh_key_name] || Chef::Config[:knife][:aws_ssh_key_id],
```

Comment:
yes...I will do the required changes to make name consistency between :aws_ssh_key_id and :ssh_key_name option, along with deprecation message.